server/pulp/server/async/tasks.py

```diff
@@ -255,7 +255,7 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         # To avoid the race condition where __call__ method below is called before
         # this change is propagated to all db nodes, using an 'upsert' here and setting
         # the task state to 'waiting' only on an insert.
-        task_status.save(fields_to_set_on_insert=['state', 'start_time'])
+        task_status.save(fields_to_set_on_insert=['state', 'start_time'], safe=True)
```

Comment:
@rbarlow Since everything is converting to mongoengine, anything we do needs to work with mongoengine. How do we set write_concern in one place with mongoengine?

I figured `write_concern={fsync: True}` would not override the configured write concern as set by the user. I figured telling mongoDB to perform a `fsync` and specifying the value for `w` would be independent options. It is possible it happens the way you describe, but I don't think either of us know for sure. We both want it safe though! Maybe @dkliban can come up with more info.